src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Abstractions/IGeneratedSymbolMacro.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Core.Contracts;
+
+namespace Microsoft.TemplateEngine.Orchestrator.RunnableProjects.Abstractions
+{
+    /// <summary>
+    /// An interface for macro created via generated symbols.
+    /// </summary>
+    public interface IGeneratedSymbolMacro : IMacro
```

Comment:
Should this actually be?:

```
public interface IGeneratedSymbolMacro : IMacro<IGeneratedSymbolConfig> { }
```

Reply:
We can drop `IGeneratedSymbolMacro<T>` and do that.

I wanted to intentionally split those
- `IMacro<T>` (T is config) where config can be created from code and evaluated.
- `IGeneratedSymbolMacro` which has `IGeneratedSymbolConfig` and can be directly evaluated; however internally it creates T from `IGeneratedSymbolConfig` and then evaluates T - just a shortcut for
```
T config = CreateConfig(IEngineEnvironmentSettings, IGeneratedSymbolConfig).
Evaluate(IGeneratedSymbolConfig, IVariableCollection, T).
```
i.e T is not really a IGeneratedSymbolConfig here.
- `IGeneratedSymbolMacro<T>` which creates T from `IGeneratedSymbolConfig` for further evaluation (not used at the moment) using `Evaluate`.

We can drop the last one if not needed and simplify `IGeneratedSymbolConfig` to be just `IMacro`<IGeneratedSymbolConfig> if there is no case where we need `IGeneratedSymbolMacro<T>`, i.e. 2 step processing.

Reply:
Decided to drop `IGeneratedSymbolMacro<T>` - we don't need it now.